### PR TITLE
[FLINK-19036][docs-zh] Translate page 'Application Profiling & Debugging' of 'Debugging & Monitoring' into Chinese

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

*Translate page 'Application Profiling & Debugging' of 'Debugging & Monitoring' into Chinese*

## Brief change log

*Translate page 'Application Profiling & Debugging' of 'Debugging & Monitoring' into Chinese*
  - The markdown file location: flink/docs/monitoring/application_profiling.zh.md
  - The page url is https://ci.apache.org/projects/flink/flink-docs-release-1.11/monitoring/application_profiling.html

## Verifying this change

*A pure translation work in documentation module.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented?  not applicable

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi @xccui 
Could you help me to review this PR if you have free time?
Thank you.

**Comment by RocMarshal** [comment]
@wuchong 
Could you help me to review this PR if you have free time?
Thank you.

File: docs/monitoring/application_profiling.zh.md [hunk 1]
```diff
@@ -25,58 +25,56 @@ under the License.
...
 
 {% highlight yaml %}
 env.java.opts: "-XX:+UnlockDiagnosticVMOptions -XX:+TraceClassLoading -XX:+LogCompilation -XX:LogFile=${FLINK_LOG_PREFIX}.jit -XX:+PrintAssembly"
 {% endhighlight %}
 
-## Analyzing Out of Memory Problems
+<a name="analyzing-out-of-memory-problems"></a>
 
-If you encounter `OutOfMemoryExceptions` with your Flink application, then it is a good idea to enable heap dumps on out of memory errors.
+## 分析内存不足问题（Out of Memory Problems）
```
Comment (xccui) [inline docs/monitoring/application_profiling.zh.md:56]: 内存不足 -> 内存溢出

File: docs/monitoring/application_profiling.zh.md [hunk 2]
```diff
@@ -25,58 +25,56 @@ under the License.
...
+
+<a name="analyzing-memory--garbage-collection-behaviour"></a>
 
-## Analyzing Memory & Garbage Collection Behaviour
+## 分析内存和 Garbage Collection
 
-Memory usage and garbage collection can have a profound impact on your application.
-The effects can range from slight performance degradation to a complete cluster failure if the GC pauses are too long.
-If you want to better understand the memory and GC behaviour of your application, then you can enable memory logging on the `TaskManagers`.
+内存使用和 garbage collection 会对你的应用程序产生巨大的影响。如果 GC 停顿时间过长，其影响范围可能从轻微的性能下降到完全的集群故障。如果你想更好地理解应用程序的内存和 GC 行为，那么可以在 `TaskManagers` 上启用内存日志记录。
```
Comment (xccui) [inline docs/monitoring/application_profiling.zh.md:70]: * 其影响范围可能从轻微的性能下降到完全的集群故障 -> 其影响力小到性能下降，大到集群全面瘫痪
* ~~那么~~

File: docs/monitoring/application_profiling.zh.md [hunk 3]
```diff
@@ -25,58 +25,56 @@ under the License.
...
 
 {% highlight yaml %}
 taskmanager.debug.memory.log: true
 taskmanager.debug.memory.log-interval: 10000 // 10s interval
 {% endhighlight %}
 
-If you are interested in more detailed GC statistics, then you can activate the JVM's GC logging via:
+如果你对更详细的 GC 统计数据感兴趣，则可以通过以下方式激活 JVM 的 GC 日志记录：
```
Comment (xccui) [inline docs/monitoring/application_profiling.zh.md:77]: * 如果你对更详细的 GC 统计数据感兴趣 -> 如果你想了解更详细的 GC 统计数据
* ~~则~~

File: docs/monitoring/application_profiling.zh.md [hunk 4]
```diff
@@ -25,58 +25,56 @@ under the License.
...
-## Overview of Custom Logging with Apache Flink
+<a name="overview-of-custom-logging-with-apache-flink"></a>
 
-Each standalone JobManager, TaskManager, HistoryServer, and ZooKeeper daemon redirects `stdout` and `stderr` to a file
-with a `.out` filename suffix and writes internal logging to a file with a `.log` suffix. Java options configured by the
-user in `env.java.opts`, `env.java.opts.jobmanager`, `env.java.opts.taskmanager`, `env.java.opts.historyserver` and
-`env.java.opts.client` can likewise define log files with
-use of the script variable `FLINK_LOG_PREFIX` and by enclosing the options in double quotes for late evaluation. Log files
-using `FLINK_LOG_PREFIX` are rotated along with the default `.out` and `.log` files.
+## 使用 Apache Flink 自定义日志概述
```
Comment (RocMarshal) [inline docs/monitoring/application_profiling.zh.md:30]:
```suggestion
## Apache Flink 自定义日志概述
```

File: docs/monitoring/application_profiling.zh.md [hunk 5]
```diff
@@ -25,58 +25,56 @@ under the License.
...
 
-## Profiling with Java Flight Recorder
+每个独立的 JobManager，TaskManager，HistoryServer，ZooKeeper 后台进程都将 `stdout` 和 `stderr` 重定向到 `.out` 文件名后缀的文件，并将其内部的日志记录写入到 `.log` 后缀的文件。用户可以在 `env.java.opts`，`env.java.opts.jobmanager`，`env.java.opts.taskmanager`，`env.java.opts.historyserver` 和 `env.java.opts.client` 配置项中配置 Java 选项（包括 log 相关的选项），同样也可以使用脚本变量 `FLINK_LOG_PREFIX` 定义日志文件，并将选项括在双引号中以供后期使用。日志文件将使用 `FLINK_LOG_PREFIX` 与默认的 `.out` 和 `.log` 后缀一起滚动。
```
Comment (RocMarshal) [inline docs/monitoring/application_profiling.zh.md:32]:
```suggestion
每个独立的 JobManager，TaskManager，HistoryServer，ZooKeeper 守护进程都将 `stdout` 和 `stderr` 重定向到名称后缀为 `.out` 的文件，并将其内部的日志记录写入到 `.log` 后缀的文件。用户可以在 `env.java.opts`，`env.java.opts.jobmanager`，`env.java.opts.taskmanager`，`env.java.opts.historyserver` 和 `env.java.opts.client` 配置项中配置 Java 选项（包括 log 相关的选项），同样也可以使用脚本变量 `FLINK_LOG_PREFIX` 定义日志文件，并将选项括在双引号中以供后期使用。日志文件将使用 `FLINK_LOG_PREFIX` 与默认的 `.out` 和 `.log` 后缀一起滚动。
```

File: docs/monitoring/application_profiling.zh.md (hunk 1 above)
Comment (RocMarshal) [inline docs/monitoring/application_profiling.zh.md:56]:
```suggestion
## 分析内存溢出问题（Out of Memory Problems）
```

File: docs/monitoring/application_profiling.zh.md (hunk 2 above)
Comment (RocMarshal) [inline docs/monitoring/application_profiling.zh.md:70]:
```suggestion
内存使用和 garbage collection 会对你的应用程序产生巨大的影响。如果 GC 停顿时间过长，其影响力小到性能下降，大到集群全面瘫痪。如果你想更好地理解应用程序的内存和 GC 行为，可以在 `TaskManagers` 上启用内存日志记录。
```

File: docs/monitoring/application_profiling.zh.md (hunk 3 above)
Comment (RocMarshal) [inline docs/monitoring/application_profiling.zh.md:77]:
```suggestion
如果你想了解更详细的 GC 统计数据，可以通过以下方式激活 JVM 的 GC 日志记录：
```

File: docs/monitoring/application_profiling.zh.md [hunk 6]
```diff
@@ -25,58 +25,56 @@ under the License.
...
 env.java.opts: "-XX:+UnlockDiagnosticVMOptions -XX:+TraceClassLoading -XX:+LogCompilation -XX:LogFile=${FLINK_LOG_PREFIX}.jit -XX:+PrintAssembly"
 {% endhighlight %}
 
-## Analyzing Out of Memory Problems
+<a name="analyzing-out-of-memory-problems"></a>
 
-If you encounter `OutOfMemoryExceptions` with your Flink application, then it is a good idea to enable heap dumps on out of memory errors.
+## 分析内存溢出问题（Out of Memory Problems）
+
+如果你的 Flink 应用程序遇到 `OutOfMemoryExceptions` ，那么启用在内存不足错误时堆转储是一个好主意。
```
Comment (RocMarshal) [inline docs/monitoring/application_profiling.zh.md:58]:
```suggestion
如果你的 Flink 应用程序遇到 `OutOfMemoryExceptions` ，那么启用在内存溢出错误时堆转储是一个好主意。
```

**Comment by RocMarshal** [comment]
@xccui 
Thank you so much for your help and suggestions. And I made some changes based on your comments.
It's hoped it could somewhat improve further.

**Comment by RocMarshal** [comment]
ping @wuchong @dianfu @alpinegizmo

**Review by dianfu — APPROVED** [review]
@RocMarshal Thanks for work and thanks @xccui for the thorough review. LGTM. Merging...

**Pull request closed without merging**